probe-rs/src/probe/stlink/mod.rs

```diff
@@ -12,17 +12,20 @@ use crate::{
         },
         communication_interface::{ArmCommunicationInterfaceState, ArmProbeInterface},
         dp::{DPAccess, DPBankSel, DPRegister, DebugPortError, Select},
-        memory::Component,
+        memory::{adi_v5_memory_interface::ArmProbe, Component},
         ApInformation, ArmChipInfo, SwoAccess, SwoConfig, SwoMode,
     },
-    DebugProbeSelector, Error as ProbeRsError, Memory, MemoryInterface, Probe,
+    DebugProbeSelector, Error as ProbeRsError, Memory, Probe,
 };
 use constants::{commands, JTagFrequencyToDivider, Mode, Status, SwdFrequencyToDelayCount};
 use scroll::{Pread, Pwrite, BE, LE};
 use std::{cmp::Ordering, convert::TryInto, time::Duration};
 use thiserror::Error;
 use usb_interface::TIMEOUT;
 
+const STLINK_MAX_READ_LEN: usize = 6144;
```

Comment:
READ and WRITE are different? Seems very odd to me ...

Reply:
This is what testing it why my ST-Link shows me, I agree that it's a bit odd...